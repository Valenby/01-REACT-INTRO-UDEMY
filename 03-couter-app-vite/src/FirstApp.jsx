//con las proptypes definimos el tipo de estas
import PropTypes from 'prop-types';

//componente hijo de props; del <ReactStricModo-main.jsx>
//destructuramos el title de las props de chrome.
export const FirstApp = ( {title, subTitle, name}) => {   //propiedades

  

  return (
    <> 
    {/* imprimimos el title, desde el navChrom*/}
      <div data-testid="test-title"> {title} </div>
      {/*<code>{ JSON.stringify (newMessage)}</code>*/}
      <p>{subTitle}</p>
      <p>{name}</p>

      {/* contenedor padre */}
    
    
        
</>
  )
}

//configuraciones a nuestros componentes(siempre al final)

//definir los tipos props
FirstApp.propTypes = {
  title: PropTypes.string.isRequired, 
  subTitle: PropTypes.string               //.isRequired
}
// no es obligatorio definirlos...
FirstApp.defaultProps = {
  name: 'valentin jeje',
  subTitle: 'no hay subtitulo',
  title: 'no hay titulo' ,
}
