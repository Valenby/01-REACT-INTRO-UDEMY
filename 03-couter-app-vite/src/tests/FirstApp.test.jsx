//con el render importamos testyin react
import { render } from '@testing-library/react';
import { FirstApp } from '../FirstApp'; //componente que estamos evaluando


describe('Pruebas en <FirstApp/>', () => { 

    //si algo cambia la purueba falla.
   //  test('debe de hacer match con el snapshot', () => { 

   //      const title = 'Hola,Soy teemo'
   //      //renderiza el componente en memoria
   //      const {container} = render(<FirstApp title={title} />) //sujeto de pruebas
        
   //      expect( container).toMatchSnapshot();
   //   })

     test('debe mostrar el titulo en h1', () => { 

        const title = 'Hola,Soy teemo'
        const {container, getByAltText} = render(<FirstApp title={title} />)
        expect( getByAltText(title) ).toBeTruthy();
     })
})