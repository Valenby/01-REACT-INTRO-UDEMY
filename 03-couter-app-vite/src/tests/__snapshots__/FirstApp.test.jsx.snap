// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <p>
     
    Hola,Soy teemo
     
  </p>
  <p>
    no hay subtitulo
  </p>
  <p>
    valentin jeje
  </p>
</div>
`;
